### ISSUE TITLE
Bogus error "lambda.param.type.incompatible"

### ISSUE THREAD

**Issue by C-D-E** [typetools/checker-framework#870]
The following program can be compiled by javac. If, however, the NullnessChecker is used, one gets 

_[9,33] error: [lambda.param.type.incompatible] incompatible parameter types in lambda expression._
```
package bar;

import java.util.stream.Stream;
import java.util.zip.ZipEntry;
import java.util.zip.ZipFile;

public class Bar {
	public static Stream<? extends ZipEntry> entries(ZipFile zipFile) {
		return zipFile.stream().filter(entry -> !entry.isDirectory() && entry.getName().endsWith(".xml"));
	}
}
```

No error is issued if you write
`(ZipEntry entry) -> !entry.isDirectory() && ...`

**Comment by mernst** [comment]
I have confirmed this bug and committed a test case as file checker/tests/nullness/java8/lambda/Issue870.java .

Thanks for the report!

**Comment by smillst** [comment]
Here's the full error message:
````
javacheck -processor nullness java8/lambda/Issue870.java -AprintVerboseGenerics -AprintAllQualifiers
java8/lambda/Issue870.java:12: error: [lambda.param.type.incompatible] incompatible parameter types in lambda expression.
                .filter(entry -> !entry.isDirectory() && entry.getName().endsWith(".xml"));
                        ^
  found   : ?[ extends @UnknownKeyFor ZipEntry super @KeyForBottom Void]
  required: capture#47 of ? extends java.util.zip.ZipEntry[ extends @UnknownKeyFor ZipEntry super @KeyForBottom Void]
1 error
````
